### ISSUE TITLE
Update-ModuleManifest reverts back CompanyName and FunctionsToExport to default

### ISSUE THREAD

**Issue by mmattes** [PowerShell/PowerShellGetv2#182]
When using the following command `Update-ModuleManifest -Path $ManifestPath -ModuleVersion $NewVersion` the CompanyName and FunctionsToExport parameter gets set back to default

## Expected Behavior
`Update-ModuleManifest -Path $ManifestPath -ModuleVersion $NewVersion` should only update the ModuleVersion and should not touch any other value

## Current Behavior
`Update-ModuleManifest -Path $ManifestPath -ModuleVersion $NewVersion` also sets CompanyName from `# CompanyName = ''` to `CompanyName = 'Unkown'` and `FunctionsToExport = '*'` to `FunctionsToExport = @()`

## Steps to Reproduce (for bugs)

1. Create a Module.psd1 and set the two parameters as shown below
```
# CompanyName = ''
FunctionsToExport = '*'
```
2. Run `Update-ModuleManifest -Path Module.psd1 -ModuleVersion "1.1.1.0"`
3. The Parameters in the Module.psd1 are now 
```
CompanyName = 'Unkown'
FunctionsToExport = @()
```

## Context
I am trying to setup a deploy pipepline for PsGallery which automatically updates the version number of my module 

## Your Environment

```PowerShell

> $PSVersionTable
Name                           Value                                                                                                                                             
----                           -----                                                                                                                                             
PSVersion                      6.0.0-beta.9                                                                                                                                      
PSEdition                      Core                                                                                                                                              
GitCommitId                    v6.0.0-beta.9                                                                                                                                     
OS                             Linux 4.4.0-53-generic #74-Ubuntu SMP Fri Dec 2 15:59:10 UTC 2016                                                                                 
Platform                       Unix                                                                                                                                              
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                           
PSRemotingProtocolVersion      2.3                                                                                                                                               
SerializationVersion           1.1.0.1                                                                                                                                           
WSManStackVersion              3.0     

> Get-Module
ModuleType Version    Name                                ExportedCommands                                                                                                       
---------- -------    ----                                ----------------                                                                                                       
Script     0.1.11.0   Invoke-Install                                                                                                                                             
Manifest   3.1.0.0    Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty...}                                                        
Manifest   3.1.0.0    Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object...}                                                              
Script     1.1.6.0    PackageManagement                   {Find-Package, Find-PackageProvider, Get-Package, Get-PackageProvider...}                                              
Script     1.5.0.0    PowerShellGet                       {Find-Command, Find-DscResource, Find-Module, Find-RoleCapability...}                                                  
Script     1.2        PSReadLine                          {Get-PSReadlineKeyHandler, Get-PSReadlineOption, Remove-PSReadlineKeyHandler, Set-PSReadlineKeyHandler...}   

> Get-Module -ListAvailable PowerShellGet,PackageManagement
ModuleType Version    Name                                ExportedCommands                                                                                                       
---------- -------    ----                                ----------------                                                                                                       
Script     1.1.6.0    PackageManagement                   {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSource...}                                                 
Script     1.5.0.0    PowerShellGet                       {Install-Module, Find-Module, Save-Module, Update-Module...}      
> Get-PackageProvider
Name                     Version          DynamicOptions                                                                                                                         
----                     -------          --------------                                                                                                                         
NuGet                    2.8.5.210        Destination, ExcludeVersion, Scope, SkipDependencies, Headers, FilterOnTag, Contains, AllowPrereleaseVersions, ConfigFile, SkipValidate
PowerShellGet            1.5.0.0          PackageManagementProvider, Type, Scope, AllowClobber, SkipPublisherCheck, InstallUpdate, NoPathUpdate, Filter, Tag, Includes, DscRes...

> Get-PackageProvider -ListAvailable
Name                     Version          DynamicOptions                                                                                                                         
----                     -------          --------------                                                                                                                         
NuGet                    2.8.5.210        Destination, ExcludeVersion, Scope, SkipDependencies, Headers, FilterOnTag, Contains, AllowPrereleaseVersions, ConfigFile, SkipValidate
PowerShellGet            1.5.0.0          PackageManagementProvider, Type, Scope, AllowClobber, SkipPublisherCheck, InstallUpdate, NoPathUpdate, Filter, Tag, Includes, DscRes...
```

**Comment by bmanikm** [comment]
I agree, CompanyName should not be changed to 'Unknown'.

FunctionsToExport change is by design as Update-ModuleManifest gets the output of Test-ModuleManifest and adds the list of function names exported by the current module. The best practice is to explicitly list the exported function names in the module manifest, this also helps in improving the command discovery performance for command autoloading scenario.

**Comment by mmattes** [comment]
> FunctionsToExport change is by design as Update-ModuleManifest gets the output of Test-ModuleManifest and adds the list of function names exported by the current module.

Isnt that bad design as no one would expect that behavior when reading `Update-ModuleManifest -Path $ManifestPath -ModuleVersion $NewVersion` ?

**Comment by bmanikm** [comment]
> Isnt that bad design as no one would expect that behavior when reading Update-ModuleManifest -Path $ManifestPath -ModuleVersion $NewVersion ?

As mentioned earlier, the best practice is to explicitly list the exported function names in the module manifest, this also helps in improving the command discovery performance for command autoloading scenario. This functionality was added as per the feedback received from the community.

It looks like your module doesn't have any exported functions, right? that's why `FunctionsToExport = @()` got added to the module manifest. If your module has exported functions and Update-ModuleManifest didn't add them properly, please provide the repro steps to investigate further.

**Comment by Benny1007** [comment]
I'm with @mmattes here. You're enforcing best practice. It is perfectly acceptable to use a wildcard in the FunctionsToExport value, and is widely documented in PowerShell documentation. 

An Update-ModuleManifest command fed a specific value should not update any other values in the manifest imo.

**Comment by marckassay** [comment]
@alerickson, has this been merged into PowerShellGet version 1.6.5?  Because I'm still seeing this issue with that version.

**Comment by marckassay** [comment]
@marckassay, looks like 1.6.5 last commit was on May 16th and #284 was merged into development branch on 22nd.
